inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/random_uniform/random_uniform_kernel_ref.cpp

```diff
@@ -0,0 +1,113 @@
+}
+
+}  // namespace
+
+namespace kernel_selector {
+
+JitConstants RandomUniformKernelRef::GetJitConstants(const random_uniform_params &params) const {
+    JitConstants jit = MakeBaseParamsJitConstants(params);
+    // When both seed values are equal to zero RandomUniform should generate non-deterministic sequence.
+    if (params.global_seed == 0 && params.op_seed == 0) {
+        std::srand(std::time(nullptr));
+        jit.AddConstant(MakeJitConstant("GLOBAL_SEED", std::rand()));
+    } else {
+        jit.AddConstant(MakeJitConstant("GLOBAL_SEED", params.global_seed));
+    }
+
+    jit.AddConstant(MakeJitConstant("OP_SEED", params.op_seed));
+    jit.AddConstant(MakeJitConstant("OUTPUT_SHAPE_SIZE", GetFullShapeSize(params.output)));
+    jit.AddConstant(MakeJitConstant("OUTPUT_STEP", getStep(params)));
+    return jit;
+}
+
+
+KernelsData RandomUniformKernelRef::GetKernelsData(const Params &params, const optional_params &options) const {
+    if (!Validate(params, options)) {
+        return {};
+    }
+
+    KernelData kernel_data = KernelData::Default<random_uniform_params>(params);
+    const random_uniform_params &new_params = dynamic_cast<const random_uniform_params &>(*kernel_data.params.get());
+
+    auto dispatch_data = SetDefault(new_params, options);
+    auto entry_point = GetEntryPoint(kernelName, new_params.layerID, params, options);
+
+    auto random_uniform_specific_jit = GetJitConstants(new_params);
+    auto jit = CreateJit(kernelName, random_uniform_specific_jit, entry_point);
+
+    FillCLKernelData(kernel_data.kernels[0], dispatch_data, params.engineInfo, kernelName, jit, entry_point, "", false,
+                     false, 3);
+
+    return {kernel_data};
+}
+
+KernelsPriority RandomUniformKernelRef::GetKernelsPriority(const Params & /*params*/,
+                                                           const optional_params & /*options*/) const {
+    return FORCE_PRIORITY_1;
+}
+
+ParamsKey RandomUniformKernelRef::GetSupportedKey() const {
+    ParamsKey k;
+    k.EnableInputDataType(Datatype::INT32);
+    k.EnableInputDataType(Datatype::INT64);
+    k.EnableInputDataType(Datatype::F16);
+    k.EnableInputDataType(Datatype::F32);
+
+    k.EnableOutputDataType(Datatype::INT32);
+    k.EnableOutputDataType(Datatype::INT64);
+    k.EnableOutputDataType(Datatype::F16);
+    k.EnableOutputDataType(Datatype::F32);
+    k.EnableDifferentTypes();
+    k.EnableAllInputLayout();
+    k.EnableAllOutputLayout();
+    k.EnableTensorOffset();
+    k.EnableTensorPitches();
+    k.EnableBatching();
+    return k;
+}
+
+bool RandomUniformKernelRef::Validate(const Params &params, const optional_params &optionalParams) const {
+    if (params.GetType() != KernelType::RANDOM_UNIFORM || optionalParams.GetType() != KernelType::RANDOM_UNIFORM) {
+        return false;
+    }
+
+    const random_uniform_params &randomUniformParams = dynamic_cast<const random_uniform_params &>(params);
+    if (randomUniformParams.inputs.empty()) {
```

Comment:
Number of passed inputs validation would be more useful.

Reply:
done